docs/git_commit_messages_requirements.md

```diff
@@ -0,0 +1,115 @@
+# Git Commit Message Requirements
+
+## 1. Use the body to explain what and why vs. how
```

Comment:
Can we rephrase this line? I feel like it's a bit confusing. Maybe something like `Use the body to explain why and on a high level what`